shenyu-plugin/shenyu-plugin-alibaba-dubbo/src/main/java/org/apache/shenyu/plugin/alibaba/dubbo/AlibabaDubboPlugin.java

```diff
@@ -74,14 +78,35 @@ public AlibabaDubboPlugin(final AlibabaDubboProxyService alibabaDubboProxyServic
             Object error = ShenyuResultWrap.error(ShenyuResultEnum.DUBBO_HAVE_BODY_PARAM.getCode(), ShenyuResultEnum.DUBBO_HAVE_BODY_PARAM.getMsg(), null);
             return WebFluxResultUtils.result(exchange, error);
         }
-        Object result = alibabaDubboProxyService.genericInvoker(param, metaData);
-        if (Objects.nonNull(result)) {
-            exchange.getAttributes().put(Constants.RPC_RESULT, result);
-        } else {
-            exchange.getAttributes().put(Constants.RPC_RESULT, Constants.DUBBO_RPC_RESULT_EMPTY);
-        }
-        exchange.getAttributes().put(Constants.CLIENT_RESPONSE_RESULT_TYPE, ResultEnum.SUCCESS.getName());
-        return chain.execute(exchange);
+
+        return Mono.create(monoSink -> {
+            ResponseFuture future = alibabaDubboProxyService.genericInvoker(param, metaData);
```

Comment:
whats version for alibaba -dubbo?

Reply:
I've tested on v2.6.5, and check the source of dubbo 2.6.6/2.6.7/2.6.8/2.6.9/2.6.10

Reply:
2.5.X ？   Can you judge his version?

1.  <  v2.6.5    also used this

```
Object result = alibabaDubboProxyService.genericInvoker(param, metaData);
```

2.  >= v2.6.5  uesd : 

```
ResponseFuture future = alibabaDubboProxyService.genericInvoker(param, metaData);
```

Reply:
I've checked the dubbo v2.5.x source,  also I think the soul alibaba-plugin was designed to work with this old version dubbo environment, and the new async implementation of this PR was also compatible with the old dubbo version. 
Ref: https://github.com/apache/dubbo/blob/2.5.x/dubbo-rpc/dubbo-rpc-default/src/main/java/com/alibaba/dubbo/rpc/protocol/dubbo/DubboInvoker.java#L90